File: LanguAI.Backend/LanguAI.Backend/Core/Models/Topic.cs
using LanguAI.Backend.Core.Enums;
using System.ComponentModel.DataAnnotations;

namespace LanguAI.Backend.Core.Models;

public class Topic
{
    [Required]
    public int Id { get; set; }

    [Required]
    public LanguageLevelEnum LanguageLevel { get; set; }

    [Required]
    public string Name { get; set; }

    [Required]
    public string NameInHun {  get; set; }

    [Required]
    public string Description { get; set; }

    [Required]
    public string DescriptionInHun { get; set; }

    public virtual ICollection<CardList> CardLists { get; set; } = new List<CardList>();
}

File: LanguAI.Frontend/src/api/models/topic.ts
/* tslint:disable */
/* eslint-disable */
import { CardList } from '../models/card-list';
import { LanguageLevelEnum } from '../models/language-level-enum';
export interface Topic {
  cardLists?: Array<CardList> | null;
  description: string;
  descriptionInHun: string;
  id: number;
  languageLevel: LanguageLevelEnum;
  name: string;
  nameInHun: string;
}
